harambe/parser/expression/evaluator.py

```diff
@@ -0,0 +1,107 @@
+from functools import wraps
+from typing import Any
+
+
+class ExpressionEvaluator:
+    functions = {}
+
+    @classmethod
+    def register(cls, func_name: str):
+        def decorator(func):
+            @wraps(func)
+            def wrapper(*args, **kwargs):
+                return func(*args, **kwargs)
+
+            cls.functions[func_name.upper()] = wrapper
+            return wrapper
+
+        return decorator
+
+    @staticmethod
+    def evaluate(expression: str, obj: Any) -> Any:
+        expression = expression.strip()
+
+        if not expression:
+            raise ValueError("Empty expression")
+
+        if "(" not in expression and ")" not in expression:
+            return ExpressionEvaluator._get_field_value(expression, obj)
+
+        func_name = expression.split("(", 1)[0].strip().upper()
```

Comment:
![image](https://github.com/user-attachments/assets/de5e3580-2948-4470-9a88-c81a1fae5e00)

will need to limit key characters first lolol. The above is a valid schema key right now which will fuk this all up

Reply:
yeah will do that as a follow up.